crates/kubelet/src/kubelet.rs

```diff
@@ -54,98 +54,77 @@ impl<T: 'static + Provider + Sync + Send> Kubelet<T> {
     /// events, which it will handle.
     pub async fn start(&self) -> anyhow::Result<()> {
         let client = kube::Client::new(self.kube_config.clone());
+
         // Create the node. If it already exists, "adopt" the node definition
         create_node(&client, &self.config, self.provider.clone()).await;
 
-        // Get the node name for use in the update loop
-        let node_name = self.config.node_name.clone();
+        // Flag to indicate graceful shutdown has started.
+        let signal = Arc::new(AtomicBool::new(false));
+        signal_hook::flag::register(signal_hook::SIGINT, Arc::clone(&signal))?;
```

Comment:
Since we are alerting only on SIGINT, we should probably just leverage `tokio::signal::ctrl_c`

Reply:
I've used ctrl_c in the past, but we may wish to support other signals such as SIGTERM, if, for instance, we wish to run Krustlet inside of a container.

Reply:
Yeah, I figure we could, although windows support for that is a bit hairy.

Also, we may want to use the `tokio-support` feature to convert this into a `Stream`

Reply:
Ah, I see.